SAT/MinisatInterfacing.cpp

```diff
@@ -40,30 +40,26 @@ unsigned MinisatInterfacing::newVar()
   return minisatVar2Vampire(_solver.newVar());
 }
 
-SATSolver::Status MinisatInterfacing::solveUnderAssumptions(const SATLiteralStack& assumps, unsigned conflictCountLimit)
+SATSolver::Status MinisatInterfacing::solveUnderAssumptionsLimited(const SATLiteralStack& assumps, unsigned conflictCountLimit)
 {
-  ASS(!hasAssumptions());
-
   // load assumptions:
+  _assumptions.clear();
   SATLiteralStack::ConstIterator it(assumps);
   while (it.hasNext()) {
     _assumptions.push(vampireLit2Minisat(it.next()));
   }
 
   solveModuloAssumptionsAndSetStatus(conflictCountLimit);
+  return _status;
+}
 
-  if (_status == Status::UNSATISFIABLE) {
-    // unload minisat's internal conflict clause to _failedAssumptionBuffer
-    _failedAssumptionBuffer.reset();
-    Minisat::LSet& conflict = _solver.conflict;
-    for (int i = 0; i < conflict.size(); i++) {
-      _failedAssumptionBuffer.push(minisatLit2Vampire(conflict[i]).opposite());
-    }
-  }
-
-  _assumptions.clear();
+SATLiteralStack MinisatInterfacing::failedAssumptions() {
+  ASS_EQ(_status, Status::UNSATISFIABLE)
 
-  return _status;
+  SATLiteralStack result;
```

Comment:
Aren't we afraid this could be more expensive than return a references to the `_failedAssumptionBuffer`?

Betting on RWO and fast allocation instead?

Reply:
It could be, but (i) this is safer and (ii) hopefully one is only calling `failedAssumptions` after at least one call to `solve`, so I'd argue it's not the biggest deal.